cli/metrics/metrics.go

```diff
@@ -89,3 +97,18 @@ func GetCommand(args []string) string {
 	}
 	return result
 }
+
+type Metadata struct {
+	Build BuildMetadata `json:"build,omitempty"`
+}
+
+// GetMetadata returns the metadata linked to the invoked command
+func GetMetadata(command string, args []string) Metadata {
+	var m Metadata
+	if command == "build" || command == "buildx" {
+		if bm := getBuildMetadata(command, args); (BuildMetadata{}) != bm {
```

Comment:
That looks a bit dangerous to compare a struct by value, this means at this location of the code we know that it’s safe (e.g. no pointer in the struct). Maybe a pointer would be safer?

Reply:
Ok I changed this behavior so we don't need to rely on comparison now.